sys/arduino/include/pins_arduino.h

```diff
@@ -0,0 +1,142 @@
+/*
+  pins_arduino.h - Pin definition functions for Arduino
+  Part of Arduino - http://www.arduino.cc/
+
+  Copyright (c) 2007 David A. Mellis
+
+  This library is free software; you can redistribute it and/or
+  modify it under the terms of the GNU Lesser General Public
+  License as published by the Free Software Foundation; either
+  version 2.1 of the License, or (at your option) any later version.
+
+  This library is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+  Lesser General Public License for more details.
+
+  You should have received a copy of the GNU Lesser General
+  Public License along with this library; if not, write to the
+  Free Software Foundation, Inc., 59 Temple Place, Suite 330,
+  Boston, MA  02111-1307  USA
+
+  NOTE: The code in this file comes from https://github.com/arduino/ArduinoCore-avr
+  (see ArduinoCore-avr/variants/.../pins_arduino.h)
```

Comment:
Note that the macros below are in fact specific to AVR boards, as they explicitly rely on implementation details of the AVR peripherals (e.g. name of memory-mapped I/O registers and their semantics).

I fully agree that having these makes sense to increase compatibility with (AVR specific) Arduino libraries. But IMO they cannot be placed in `sys/arudino`, as `sys/...` is reserved for platform independent stuff. Maybe place these into `cpu/atmega_common/include/arduino_arch.hpp` and include that from `sys/arduino/include/arduino.hpp`, e.g. via

```C
#ifdef MODULE_ATMEGA_COMMON
#include "arduino_arch.hpp"
#endif
```
?